### DISCUSSION TITLE
Mismatch of values between mySigen app and modbus integration

### DISCUSSION THREAD

**Discussion by michaelroper** [TypQxQ/Sigenergy-Home-Assistant-Integration#38] [General]
Hi - wondering if anyone can help explain why I'm getting different values between what's in the mySigen app (on the right) and the values produced by the sensors in this integration? Am I comparing the wrong values?

![image](https://github.com/user-attachments/assets/78eadb0c-84d0-42c1-9f20-e417528abc3a)

**Comment by TypQxQ** [comment]
Hi!
All the daily values are calculated by HomeAssistant because they are not provided by Sigenergy trough Modbus. 
On the default dashboard example you can find the calculated values on the last tab.
Differences are therefore inevitable. 
Which one is more accurate towards what your grid provider charges?
Hope this helps. 
/Andrei

**Reply by michaelroper** [reply]
Thanks Andrei - that last tab is where I grabbed the screenshot from. I realise the values are being calculated on multiple factors, which obviously makes it less likely to match exactly - it's odd that they don't provide those values in the modbus data! I guess "daily" data depends on things like users/system timezone, so it's not a discrete value..

The data that I can get from my provider is 48 hours behind, so I'll have to check and compare todays data in a couple days. Even then, I will only be able to compare the grid import/export values, not the generation/consumption ones..

Cheers,
Michael

**Reply by michaelroper** [reply]
Hey Andrei - I grabbed a few days of grid import data from my energy provider, and compared to the values from the mySigen app as well as the ModBus integration calculated value... the calculated value is definitely not in the same range...

Date | Provider | mySigen App | Integration
-- | -- | -- | --
2024-11-21 | 0.253 | 0.23 | 1.02
2024-11-22 | 0.111 | 0 | 0.27
2024-11-23 | 0.103 | 0 | 0.21
2024-11-24 | 0.25 | 0.26 | 4.87
2024-11-25 | 0.159 | 0.06 | 1.33

And you can also see from the history chart of the calculated import value - on some days it seems to do a big jump that doesn't necessarily correspond with any large grid import..

![image](https://github.com/user-attachments/assets/3bf20e95-097c-4f37-8d61-3c2120d5eb72)

**Comment by TypQxQ** [comment]
I checked on my system that I updated yesterday to newest firmware. The day before yesterday HA reported 35.08KW Sigenergy 35.38 and my grid provider 36.6kW of import from from grid.

Are you using the latest modbus code?
What versions of HA and Sigenergy do you use?
Check how the "Sigen Grid sensor active power" at the time of the abnormality.

**Reply by michaelroper** [reply]
As far as I know - I replaced it last week before my system came online. It did say something about updating units, and I think i just pressed 'fix it' or whatever is the default. Would it be worth just removing the integration and data, and starting from scratch?

**Reply by TypQxQ** [reply]
What versions of HA and Sigenergy do you use?
Check how the "Sigen Grid sensor active power" at the time of the abnormality.

**Comment by michaelroper** [comment]
HA Core is 2024.11.3, Sigenergy app is 2.1.0, SigenStor firmware (i think?) is V100r001c21, Gateway TP is V100r001c00..

This is with the active power sensor too..
![image](https://github.com/user-attachments/assets/b3ada738-c9e4-48ed-bb2a-00421ebe293a)

**Comment by azebro** [comment]
I believe in the app the internal consumption of the unit is hidden. For me Sig internally consumes ~150W continuesly which is a lot actually. If you calculate the internal consumption and deduct from your figured, you will get to the app values. I noticed some other discrepancies and opened a ticket with Sigen but they keep closing it...

**Comment by jehoshua7** [comment]
I'm understanding that this is a discussion (not an issue tracking process), ..

Whilst I'm not using the ModBus, I have noticed a mismatch of values from MySigen/SigenCloud and the smartmeter. Quite a lot. Two questions please:

1. Is the data stored in the inverter or in some server on the internet ?
2. We have it all set for 'off grid' even though we are still connected to the grid. SigEnergy keeps reported our power draw from the grid as zero, yet every few days the smartmeter will show 1 kW used. Recently a comment - "It's normal to use a tiny bit of power from the grid mainly when you have a sudden demand for power however that will usually only be for a few seconds." , so that seems to confirm that despite the 'off grid' setup, it still does occasionally  draw power from the grid.

**Comment by TypQxQ** [comment]
Please see below link for posible fix and discusion:
https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration/issues/57
